scripts/prepare_for_registration_bulk_creation.py

```diff
@@ -0,0 +1,117 @@
+        return handle_error(initiator, inform_product=True)
+
+    schema_id = parsing_output.get('schema_id', None)
+    try:
+        schema = RegistrationSchema.objects.get(_id=schema_id)
+    except RegistrationProvider.DoesNotExist or RegistrationProvider.MultipleObjectsReturned:
+        logger.error('Registration schema not found or multiple results returned: [_id={}] '.format(schema_id))
+        return handle_error(initiator, inform_product=True)
+
+    upload = RegistrationBulkUploadJob.create(payload_hash, initiator, provider, schema)
+    logger.info('Inserting the bulk upload job [hash={}] into table '
+                'RegistrationBulkUploadJob ...'.format(upload.payload_hash))
+    if not dry_run:
+        try:
+            upload.save()
+        except ValidationError as e:
+            logger.error('Insertion failed: [error={}, message={}]'.format(type(e), e.messages))
+            return handle_error(initiator, inform_product=True)
+        upload.reload()
+        logger.info('Insertion successful: [pk={}, hash={}] '.format(upload.id, upload.payload_hash))
+    else:
+        logger.info('Dry run: insertion not saved')
+
+    registration_rows = parsing_output.get('registrations', [])
+    if not registration_rows:
+        logger.error('Missing registration rows')
+        return handle_error(initiator, inform_product=True)
+
+    logger.info('Preparing [{}] registration rows for bulk creation ...'.format(len(registration_rows)))
+    bulk_upload_rows = []
+    for registration_row in registration_rows:
+        bulk_upload_row = RegistrationBulkUploadRow(upload=upload, draft_registration=None, is_completed=False,
+                                                    is_picked_up=False, csv_raw=registration_row.get('csv_raw', ''),
+                                                    csv_parsed=registration_row.get('csv_parsed'))
+        bulk_upload_rows.append(bulk_upload_row)
+
+    if dry_run:
+        logger.info('Dry run: bulk insertion not run')
+        logger.info("Dry run: complete")
+        return
+
+    try:
+        logger.info('Bulk inserting [{}] registration rows into table '
+                    'RegistrationBulkUploadRow ...'.format(len(bulk_upload_rows)))
+        created_objects = RegistrationBulkUploadRow.objects.bulk_create(bulk_upload_rows)
+    except (ValueError, IntegrityError) as e:
+        logger.error('Bulk insertion failed: [error={}, cause=\n{}\n]'.format(type(e), e.__cause__))
+        return handle_error(initiator, inform_product=True)
+    logger.info('[{}] rows successfully inserted.'.format(len(created_objects)))
+
+    logger.info('Updating job state ...')
+    upload.state = JobState.INITIALIZED
+    try:
+        upload.save()
+    except ValidationError as e:
+        logger.error('Job state update failed: [error={}, message={}]'.format(type(e), e.messages))
+        return handle_error(initiator, inform_product=True)
+    logger.info('Job state updated')
+    upload.reload()
```

Comment:
Do you really need to reload after `save()` to get the updated state of the model instance?

Reply:
No, I don't. `.reload()` is usually used after insertion to grab the `pk`. I will remove this.